doc/chapters/api/index.rst

```diff
@@ -32,102 +32,187 @@ BindMount
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
-        :pathInHost: ``string`` absolute path in the host.
-        :pathInContainer: ``string`` the absolute path to mount to in the container.
-        :readOnly: ``bool`` indicates whether the directory is read-only or not.
-
-|
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
+* pathInHost: ``string`` absolute path in the host.
+* pathInContainer: ``string`` the absolute path to mount to in the container.
+* readOnly: ``bool`` indicates whether the directory is read-only or not.
+
+Error sources
+#############
+* Bad state: The container is not started or ready
+* Bad path
+    * The host path does not exist
+    * The destination path is not absolute
+    * The destination path is already mounted to
+* Filesystem issues
+    * Could not create parent directories
+    * Could not create the target directory (if mounting a directory)
+    * Could not touch the target file (if mounting a file)
+* Mount issues
+    * Mount failed (:linuxman:`mount(2)` failed)
+    * Couldn't move the mount
+    * Failed to re-mount read-only
+
+Currently, failing mounts are not rolled back.
 
 Create
 ------
 Creates a container with given configuration.
 
-:Parameters:
-        :config: ``string`` config file in json format.
+Parameters
+##########
+* config: ``string`` config file in json format.
+
+Example config JSON::
 
-                Example config JSON::
+[{"enableWriteBuffer": true}]
 
-                [{"enableWriteBuffer": true}]
+Return Values
+#############
+* containerID: ``int32`` ID of created SoftwareContainer.
 
-|
+Error sources
+#############
+* Network setup issues
+    * Bridge device not available and unable to set it up
+    * Given netmask can not be used to assign IP address to container
 
-:Return Values:
-        :containerID: ``int32`` ID of created SoftwareContainer.
+* LXC issues
+    * Couldn't prepare file system while initializing container
+    * Container ID already being in use (caused by lingering old containers)
+    * LXC config couldn't be loaded
+    * LXC could not create container
 
 Destroy
 -------
 Tears down all active gateways related to container and shuts down the
 container with all reserved sources.
 
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
+
+Error sources
+#############
+* Invalid ID: No matching container exists.
+* Invalid state: The container is not ready or is suspended
+* Gateway errors: One or more of the gateways might have failed to shut down
+* LXC issues:
+    * LXC could not shutdown the container
+    * LXC could not force stop the container (if shutdown fails)
+    * LXC could not destroy the container
 
-|
+Failing to destroy the container leads to it being put into an invalid state. One should
+(ironically) destroy such a container.
```

Comment:
This is ironic but not very helpful. I think we should skip the "ironically" and mention what possibilities the user have in this case, or perhaps better, link to some part of the docs that describe what an unrecoverable state means (e.g. "restart SC").